### PR TITLE
Verify module IDs

### PR DESCRIPTION
**Pull request by pamaury**
See #17605.

The goal of this PR is to add a check to enforce that module IDs are not conflicting.
The approach is to run the `opentitantool status lint` command on every relevant ELF file (see commits for details).

One downside of this approach is it triggers a build of `opentitantool` and also builds all the test ELF files. The former is quite slow because opentitanlib depends on `pqcrypto-sphincsplus` even though this is not useful for this particular check. This can also be fixed by reorganizing `opentitanlib`.

One possibility would be to run those checks at the same time as we run the device tests on FPGA since everything has to be built anyway, but this could be a bit confusing.

Questions:
- At the moment I am using an aspect with an explicit output group, but apparently bazel supports [validation actions](https://bazel.build/extending/rules#validation_actions), I don't know if it worth using those?
- While going through `quality/BUILD.bazel`, I noticed an hardcoded list `RUST_TARGETS` of targets. Could this use similar mecanism to what is done in this PR to avoid a list?

### CODE REVIEW

**Comment by pamaury** [comment]
The build failure is due to `manuf_cp_device_info_flash_wr_test_unlocked0_to_prod_end_functest_fpga_cw310_rom_with_fake_keys` which is marked as flaky

**Review by timothytrippel — APPROVED** [review]

File: ci/scripts/check-module-ids.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+set -e
+
+# List of all targets that need to be checked: we only consider targets that
+# produce a binary file and that depend on the status library. Since every test
+# is usually built for real targets and dv, we only check for one real target to
+# save time.
+query_elfs='
+    filter(
+        ".*prog_fpga_cw310.*",
```
[the remark this answers is not in the document]
Comment (pamaury) [reply ci/scripts/check-module-ids.sh:16]: Yes that's a very good point.

**Comment by timothytrippel** [comment]
> This `Check status_t Module IDs` seems to be causing errors on PRs. Is there an issue tracking this?

Is this still true? could you share some example PRs with issues?

**Comment by marnovandermaas** [comment]
> > This `Check status_t Module IDs` seems to be causing errors on PRs. Is there an issue tracking this?
> 
> Is this still true? could you share some example PRs with issues?

Yup, I've created an issue for it here: https://github.com/lowRISC/opentitan/issues/19190

**Pull request merged**